test/llilc_checkpass.py

```diff
@@ -66,62 +67,131 @@ def main(argv):
+                                        good_set[method] = True
+                                        return True, bad_count
+                                    return None, bad_count
+                                elif method in bad_set:
+                                    if not bad_set[method]:
+                                        bad_set[method] = True
+                                        return True, bad_count + 1
+                                    return None, bad_count
+                                else:
+                                    return False, bad_count
+
+                            matched_base_method = False
+                            if m.group(1) is not None:
+                                matched_base_method, target_failure_diff = update(m.group(2), base_passing, base_failing, target_failure_diff)
+                            else:
+                                matched_base_method, target_success_diff = update(m.group(4), base_failing, base_passing, target_success_diff)
+                                if matched_base_method is not None and not matched_base_method:
+                                    new_target_failures += 1
+
+                            if matched_base_method is not None:
+                                if not matched_base_method:
+                                    new_target_methods += 1
+                                else:
+                                    base_matched_methods += 1
+
+                    missing_base_methods += len(base_passing) + len(base_failing) - base_matched_methods
             except:
                 e = sys.exc_info()[0]
-                print('Error: CheckPass failed due to ', e)
+                print 'Error: CheckPass failed due to ', e
+                traceback.print_exc()
                 return const.GeneralError
         else:
             missing_result_file = os.path.join(args.diff_result_path, file)
-            print('Error: diff result does not result file ', missing_result_file)
+            print 'Error: diff directory does not have result file ', missing_result_file
             return const.MissingResult
 
-    print('CheckPass: ', count_new_failed, ' test cases passed in base result but failed in diff result.')
-    print('CheckPass: ', count_new_passed, ' test cases failed in base result but passed in diff result.') 
-    if count_new_failed == 0:
-        print('CheckPass Passed.')
-        return const.CheckPassOK
+    print 'CheckPass:', target_failure_diff, 'method(s) compiled successfully in base result that failed in target result.'
+    print 'CheckPass:', target_success_diff, 'method(s) failed in base result that compiled successfully in target result.'
+    print 'CheckPass:', missing_base_methods, 'method(s) processed in base result that were not processed in target result.'
+    print 'CheckPass:', new_target_methods, 'method(s) not processed in base result that were processed in target result.'
```

Comment:
Seems funny to me that the other four categories you report are disjoint but this one is a superset of new_target_failures; I'd have computed and reported new_target_successes rather than new_target_methods to keep them all disjoint (and FWIW that would let you move the code on 164-168 up to 158 and make it symmetric with the code on 161-162).  But I don't think it's wrong to do it this way if you really prefer it.